### DISCUSSION TITLE
Custom functionality for copy button in CodeBlock component

### DISCUSSION THREAD

**Discussion by rmatuszczakp** [twilio-labs/paste#2689] [General] [answered]
Hello team, I have a question regarding the[ CodeBlock component ](https://paste.twilio.design/components/code-block/). The component includes its own copy button. 
![Screen Shot 2022-09-26 at 9 10 11 AM](https://user-images.githubusercontent.com/102552876/192327100-0f2614d1-a638-48bb-bc6e-a01aabcd80a7.png)
Is there a way to hide this button, or give it a custom functionality? My team currently has a copy button for our code snippets that has a special functionality of doing certain string replacements when used. For example, it will replace a by-default hidden auth token with the actual value.

I couldn't find anything about it in the docs. I guess either hiding or customizing would be good for us.

**Comment by shleewhite** [comment, marked as answer]
Hi @rmatuszczakp,

Thanks for posting this! Would a `copyTextFormatter` prop work for your use case? It would work like this:
```jsx
const myCodeBlock = () => {
  const myCode = `const id = XXXX`;
  
  return (
    <CodeBlock
      code={myCode}
      language='javascript'
      copyTextFormatter={(myCode) => {
          const formattedCode = myCode;
          // do something to the string
          return formattedCode;
      }}
   />
  )
}
```

**Reply by rmatuszczakp** [reply]
Hi @shleewhite yes that would definitely work!

**Reply by rmatuszczakp** [reply]
When do you think that could become available?

**Reply by shleewhite** [reply]
I have started a PR to add the prop: https://github.com/twilio-labs/paste/pull/2703

I will follow up again once it is merged and the new CodeBlock package is released on npm.

**Reply by nkrantz** [reply]
Hi @rmatuszczakp - the PR has been merged.
